ttyplot.c

```diff
@@ -616,17 +584,21 @@ int main(int argc, char *argv[]) {
             redraw_needed = true;
 
         // Handle signals.
-        if (sigint_pending) {
-            break;
-        }
-        if (sigwinch_pending) {
-            sigwinch_pending = 0;
-            endwin();
-            initscr();
-            erase();
-            refresh();
-            gethw();
-            redraw_needed = true;
+        if (select_ret > 0 && FD_ISSET(signals[0], &read_fds)) {
+            int signum;
+            int count = read(signals[0], &signum, sizeof(signum));
+            if (count > 0) {
+                if (signum == SIGINT) {
+                    break;
+                } else if (signum == SIGWINCH) {
```

Comment:
Just an idea: This `else` could be resolved for lower nesting level and arguably lower complexity.

Reply:
OK. Done.